src/Core/Lock/ILockDriver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Friendica\Core\Lock;
+
+/**
+ * Lock Driver Interface
+ *
+ * @author Philipp Holzer <admin@philipp.info>
+ */
+interface ILockDriver
+{
+	/**
+	 * @brief Checks, if a key is currently locked to a or my process
+	 *
+	 * @param string $key 		The name of the lock
+	 * @return bool
+	 */
+	public function isLocked($key);
+
+	/**
+	 *
+	 * @brief Acquires a lock for a given name
+	 *
+	 * @param string  $key      The Name of the lock
+	 * @param integer $timeout  Seconds until we give up
+	 *
+	 * @return boolean Was the lock successful?
+	 */
+	public function acquireLock($key, $timeout = 120);
```

Comment:
This is already a class bout Locks, this can advantageously be renamed to just `acquire`.